ReportPortal.Shared.Tests/ConfigurationTest.cs

```diff
@@ -0,0 +1,198 @@
+            Assert.Equal(new List<int> { 1, 2 }, list);
+        }
+
+        [Fact]
+        public void ShouldGetSeveralListsFromJsonFile()
+        {
+            var tempFile = Path.GetTempFileName();
+
+            File.WriteAllText(tempFile, "{\"prop1\": [\"value1\", \"value2\"], \"prop2\": [\"value11\", \"value22\"]}");
+
+            var config = new ConfigurationBuilder().AddJsonFile(filePath: tempFile).Build();
+
+            Assert.Equal(2, config.Values.Count);
+
+            var variable1 = config.GetValue<string>("prop1");
+            Assert.Equal("value1;value2;", variable1);
+
+            var variable2 = config.GetValue<string>("prop2");
+            Assert.Equal("value11;value22;", variable2);
+        }
+
+        [Fact]
+        public void ShouldRaiseExceptionIfJsonIsIncorrect()
+        {
+            var tempFile = Path.GetTempFileName();
+
+            File.WriteAllText(tempFile, "Bh}");
+
+            Assert.Throws<XmlException>(() => new ConfigurationBuilder().AddJsonFile(filePath: tempFile).Build());
+        }
+
+        [Fact]
+        public void ShouldRaiseExceptionIfVariableNotFound()
+        {
+            var config = new ConfigurationBuilder().Build();
+
+            Assert.Throws<KeyNotFoundException>(() => config.GetValue<string>("a"));
+        }
+
+        [Fact]
+        public void ShouldReturnDefaultIfVariableNotFound()
+        {
+            var config = new ConfigurationBuilder().Build();
+
+            var a = config.GetValue("a", "abc");
+
+            Assert.Equal("abc", a);
+        }
+
+        [Fact]
+        public void ShouldRaiseExceptionIfListNotFound()
+        {
+            var config = new ConfigurationBuilder().Build();
+
+            Assert.Throws<KeyNotFoundException>(() => config.GetValues<string>("a"));
+        }
+
+        [Fact]
+        public void ShouldReturnDefaultIfListNotFound()
+        {
+            var config = new ConfigurationBuilder().Build();
+
+            var list = config.GetValues("a", new List<string> { "abc" });
+
+            Assert.Equal(new List<string> { "abc" }, list);
+        }
+
+        [Fact]
+        public void ShouldMergeValuesIfStartsWithPlus()
+        {
+            Environment.SetEnvironmentVariable("REPORTPORTAL__A", "+=value1");
+
+            var config = new ConfigurationBuilder().AddEnvironmentVariables().AddEnvironmentVariables().Build();
+
+            var variable = config.GetValue<string>("A");
+
+            Assert.Equal("value1value1", variable);
+        }
+
+        [Fact]
+        public void ShouldMergeListOfValuesIfStartsWithPlus()
+        {
+            Environment.SetEnvironmentVariable("REPORTPORTAL__A", "+=value1;");
+
+            var config = new ConfigurationBuilder().AddEnvironmentVariables().AddEnvironmentVariables().Build();
+
+            var list = config.GetValues<string>("A");
+
+            Assert.Equal(new List<string> { "value1", "value1" }, list);
+        }
+    }
+}
```

Comment:
I would add a couple of tests for cases when AddJsonFile and AddEnvironmentVariables are used at the same time and override each others values depending their order.

Reply:
Done.